### ISSUE TITLE
Add ToString and arirthmetic TC instance to Time/Instant

### ISSUE THREAD

**Issue by jaschdoc** [flix/flix#5025]
Currently it's quite hard to time a function or just record and print some timestamps. There is a `Time/Instant.subtract` function but that takes an `Instant` and `Int32` but there's no way to convert `Instant` to `Int32` either so effectively there's no way to compute `t1-t0` currently.

**Comment by mlutze** [comment]
IMO it does not make sense to subtract one Instant from another. Otherwise we get weirdness like "today minus yesterday equals January 2, 1970"

Instead there should be some function finding the difference between instances that returns a `Duration`, from which you can get your milliseconds or whatever.

**Comment by mlutze** [comment]
I'm also very wary of making a ToString for `Instant` since I think it is really meant to be abstract. Maybe we should instead have functions to convert it into the format the user wants (Date, DateTime, etc) and then ToStrings for those.

**Comment by stephentetley** [comment]
> Instead there should be some function finding the difference between instances that returns a Duration, from which you can get your milliseconds or whatever.

In Java this is `Duration.between` - `Instant` implements the `Temporal` interface.

**Comment by stephentetley** [comment]
For arithmetic on time "objects" we need something a bit more than regular addition and subtraction. 

Vector space / affine space operations as per Haskell's Vector-Space package are what we want (a generic version needs some sort of MPTCs):

https://hackage.haskell.org/package/vector-space

**Comment by jaschdoc** [comment]
> I'm also very wary of making a ToString for `Instant` since I think it is really meant to be abstract. Maybe we should instead have functions to convert it into the format the user wants (Date, DateTime, etc) and then ToStrings for those.

Yes I agree with this, however, I could see a few programmers thinking "just let me print this please". Maybe the ToString instance could just convert to DateTime (whatever we decide) and use its ToString instance?

**Comment by stephentetley** [comment]
`Instant` in Java prints an ISO-8601 string for `toString` I think Flix's `Time.Instant` should use that.

**Comment by jaschdoc** [comment]
For arithmetic see

https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/time/temporal/TemporalAmount.html#addTo(java.time.temporal.Temporal)

For `ToString` see https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/time/Instant.html#toString()

**Comment by magnus-madsen** [comment]
We have removed these instances from the std lib
